src/trio/_tests/test_threads.py

```diff
@@ -895,6 +895,7 @@ async def main(use_system_task: bool) -> None:
 
 
 async def test_trio_token_weak_referenceable() -> None:
+    await _core.checkpoint()
     token = _core.current_trio_token()
     assert isinstance(token, _core.TrioToken)
     weak_reference = weakref.ref(token)
```

Comment:
~~I don't think there's any reason to enable ASYNC910/async-function-no-checkpoint on test files. Or is this a different check saying "async function with no awaits can be sync?"~~
oh this is RUF029/unused-async

Reply:
Fixed in new version